src/core/public/http/http_setup.ts

```diff
@@ -155,17 +158,28 @@ export const setup = (
       (promise, interceptor) =>
         promise.then(
           async httpResponse => {
+            current = httpResponse;
             checkHalt(controller);
 
             if (!interceptor.response) {
               return httpResponse;
             }
 
-            current = (await interceptor.response(httpResponse, controller)) || httpResponse;
-
-            return current;
+            return Object.assign(
```

Comment:
If I understand correctly, the goal of this statement is to allow the interceptor to return a `Partial<HttpResponse>`. If the interceptor doesn't return one of the properties of the `HttpResponse`, we want to use the properties from the original `httpResponse`. I don't think this does what you want:

```
> Object.assign({ response: 'new-response', body: 'new-body' }, { request: 'original-request', response: 'original-response', body: 'original-response' });
{ response: 'original-response',
  body: 'original-response',
  request: 'original-request' }
>
```

We also discussed on Slack, that we didn't see a reason why a http response interceptor should ever be changing the request itself, as the request should have already happened.